src/overlays/ovl_DF4260/fox_co.c

```diff
@@ -0,0 +1,3333 @@
+    actor->vel.x = actor->fwork[0] * temp_sin;
+    temp_cos = __cosf(actor->obj.rot.y * M_DTOR);
+    actor->vel.z = actor->fwork[0] * temp_cos;
+    Matrix_RotateY(gCalcMatrix, actor->obj.rot.y * M_DTOR, 0);
+
+    switch (actor->unk_0B8) {
+        case 3:
+            break;
+        case 0:
+            actor->timer_0BC = 40;
+            actor->unk_0B8 = 1;
+            actor->unk_0B6 = 0;
+            break;
+        case 1:
+            actor->fwork[0] = -10.0f;
+            Texture_Scroll(&D_60329C0, 16, 16, 1);
+            if (actor->timer_0BC == 0) {
+                actor->unk_0B8 = 2;
+                actor->iwork[2] = (s32) (Rand_ZeroOne() * 10.0f) + 10;
+            }
+            break;
+        case 2:
+            actor->fwork[0] = -10.0f;
+            Texture_Scroll(&D_60329C0, 16, 16, 1);
+            actor->unk_0B6++;
+            if (actor->unk_0B6 >= Animation_GetFrameCount(D_602AA04)) {
+                actor->unk_0B8 = 3;
+            }
+            if (actor->unk_0B6 == (Animation_GetFrameCount(D_602AA04) - actor->iwork[2])) {
+                actor->iwork[1] = 1;
+                temp_s1->unk_48 = 1;
+                sp54.x = 0.0f;
+                sp54.y = 0.0f;
+                sp54.z = 30.0f;
+                Matrix_MultVec3fNoTranslate(gCalcMatrix, &sp54, &sp48);
+                temp_s1->vel.x = sp48.x;
+                temp_s1->vel.y = sp48.y;
+                temp_s1->vel.z = sp48.z;
+                Audio_PlaySfx(0x2902401C, &actor->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            }
+            break;
+    }
+    if (actor->iwork[1] == 0) {
+        temp_s1->obj.pos.x = actor->fwork[2];
+        temp_s1->obj.pos.y = actor->fwork[6];
+        temp_s1->obj.pos.z = actor->fwork[10];
+        temp_s1->obj.rot.y = actor->obj.rot.y;
+        temp_s1->vel.y = 0.0f;
+    }
+    Math_SmoothStepToVec3fArray(sp60, actor->vwork, 0, Animation_GetFrameData(D_602AA04, actor->unk_0B6, sp60), 1.0f,
+                                1.0f, 1.0f);
+}
+
+extern AnimationHeader D_602A520[];
+
+void func_DF4260_8018B418(Actor* actor) {
+    s32 pad;
+    Vec3f sp54[20];
+    Object_80* temp_v0_2;
+    f32 temp_sin;
+    f32 temp_cos;
+    s32 pad2[4];
+
+    func_DF4260_8018ACE0(actor);
+    temp_sin = __sinf(actor->obj.rot.y * M_DTOR);
+    actor->vel.x = actor->fwork[0] * temp_sin;
+    temp_cos = __cosf(actor->obj.rot.y * M_DTOR);
+    actor->vel.z = actor->fwork[0] * temp_cos;
+    switch (actor->unk_0B8) {
+        case 0:
+            actor->unk_0B8 = 1;
+            break;
+        case 1:
+            actor->fwork[0] = 5.0f;
+            actor->fwork[1] += 5.0f;
+            Texture_Scroll(&D_60329C0, 16, 16, 1);
+            actor->unk_0B6++;
+            if (actor->unk_0B6 >= Animation_GetFrameCount(D_602A520)) {
+                actor->unk_0B6 = 0;
+            }
+            break;
+    }
+    if (actor->iwork[1] == 0) {
+        temp_v0_2 = &gObjects80[actor->iwork[0]];
```

Comment:
is this temp needed?

Reply:
it looks like it, maybe thers another way to match though